### PR TITLE
Fixes #24978 - support for DDNS lease entries

### PR DESCRIPTION
**Pull request by lzap**
We simply need to ignore these lines, parser was going crazy about incorrect literals.

### CODE REVIEW

**Comment by ifreislich** [comment]
With this patch the parser hasn't choked (yet) on my dhcpd.leases file.  Some of the originally offending data is in the lease file.

**Comment by lzap** [comment]
Want to take a look @ik_5? See the issue for the whole story.

**Comment by ifreislich** [comment]
Been running with this patch for 5 days.  No issues so far.

File: modules/dhcp_common/isc/configuration_parser.rb [hunk 1]
```diff
@@ -269,12 +269,18 @@ def vendor_option_space
         def option
           Rsec::Fail.reset
           keyword_option = word('option').fail 'keyword_option'
-          keyword_supersede = word('supersede').fail 'keyword_option'
+          keyword_supersede = word('supersede').fail 'keyword_supersede'
```
Comment (ik5) [inline modules/dhcp_common/isc/configuration_parser.rb:272]: nice catch

**Review by ik5 — APPROVED** [review]
LGTM and tests are working as well.

**Review by mmoll — APPROVED** [review]
looks good to me

**Comment by mmoll** [comment]
merged, díky @lzap!

File: modules/dhcp_common/isc/configuration_parser.rb (hunk 1 above)
Comment (lzap) [reply modules/dhcp_common/isc/configuration_parser.rb:272]: Now the question is what this is for? ;-) I have no idea.

Comment (ik5) [reply modules/dhcp_common/isc/configuration_parser.rb:272]: Based on: http://rsec.herokuapp.com/ref

>word(pattern) helper
>
>A word is a token wrapped with word boundaries
>
>example:
>
>```ruby
>assert_equal ['yes', '3'], seq('yes', '3').parse('yes3')
>assert_equal INVALID, seq(word('yes'), '3').parse('yes3')
>```

and
>
>fail(*tokens)
>
>When parsing fails on the preceding term, show "expect tokens" error for that tokens value.

Comment (lzap) [reply modules/dhcp_common/isc/configuration_parser.rb:272]: I felt like this is something like that :-)

**Comment by ik5** [comment]
I went and checked it out for this PR :)

File: modules/dhcp_common/isc/configuration_parser.rb (hunk 1 above)
Comment (ik5) [reply modules/dhcp_common/isc/configuration_parser.rb:272]: I checked it out prior of me approving it :P

**Pull request merged**